frontend/public/components/deploy-image.tsx

```diff
@@ -0,0 +1,439 @@
+  };
+
+  search = event => {
+    event.preventDefault();
+
+    const { namespace, imageName } = this.state;
+    const importImage = {
+      kind: 'ImageStreamImport',
+      apiVersion: 'image.openshift.io/v1',
+      metadata: {
+        name: imageName,
+        namespace: namespace
+      },
+      spec: {
+        import: false,
+        images: [{
+          from: {
+            kind: 'DockerImage',
+            name: imageName
+          }
+        }]
+      },
+      status: {}
+    };
+
+    this.setState({
+      loading: true,
+      isi: null,
+      name: '',
+      error: null,
+      searchError: null,
+    });
+
+    k8sCreate(ImageStreamImportsModel, importImage)
+      .then((imageStreamImport) => {
+        const name = _.get(imageStreamImport, 'spec.images[0].from.name');
+        const status = _.get(imageStreamImport, 'status.images[0].status');
+        if (status.status === 'Success') {
+          this.setState({
+            loading: false,
+            isi: {
+              name,
+              image: _.get(imageStreamImport, 'status.images[0].image'),
+              tag: _.get(imageStreamImport, 'status.images[0].tag'),
+              status
+            },
+            name,
+          });
+        } else {
+          this.setState({
+            loading: false,
+            searchError: _.upperFirst(status.message),
+          });
+        }
+      }, err => this.setState({error: err.message}));
+  };
+
+  save = event => {
+    event.preventDefault();
+
+    this.setState({
+      inProgress: true,
+      error: null
+    });
+
+    const { name, namespace, isi } = this.state;
+
+    const annotations = {
+      'openshift.io/generated-by': 'OpenShiftWebConsole'
+    };
+
+    const volumes = [], volumeMounts = [];
+    let volumeNumber = 0;
+    _.forEach(isi.image.dockerImageMetadata.Config.Volumes, (value, path) => {
+      volumeNumber++;
+      const volumeName = `${name}-${volumeNumber}`;
+      volumes.push({
+        name: volumeName,
+        emptyDir: {}
+      });
+      volumeMounts.push({
+        name: volumeName,
+        mountPath: path
+      });
+    });
+
+    const ports = parsePorts(isi.image);
+
+    const promises = [];
+
+    const deploymentConfig = {
+      kind: 'DeploymentConfig',
+      apiVersion: 'apps.openshift.io/v1',
+      metadata: {
+        name,
+        namespace,
+        labels: {app: name},
+        annotations
+      },
+      spec: {
+        strategy: {
+          resources: {}
+        },
+        triggers: [{
+          type: 'ConfigChange'
+        }, {
+          type: 'ImageChange',
+          imageChangeParams: {
+            automatic: true,
+            containerNames: [
+              name
+            ],
+            from: {
+              kind: 'ImageStreamTag',
+              name: `${isi.name}:${isi.tag}`,
+              namespace
+            }
+          }
+        }],
+        replicas: 1,
+        test: false,
```

Comment:
What is `test`?

Reply:
No idea.  Carried over from https://github.com/openshift/origin-web-console/blob/master/app/scripts/services/images.js#L176.  Delete it?

Reply:
Oh, hey. That's really a thing: https://github.com/openshift/api/blob/master/apps/v1/types.go#L65-L69

I'd just leave it out rather than explicitly setting it (even though we set it before...)